packages/flutter/lib/src/widgets/semantics_debugger.dart

```diff
@@ -0,0 +1,346 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:sky_services/semantics/semantics.mojom.dart' as engine;
+
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+
+class SemanticsDebugger extends StatefulComponent {
+  const SemanticsDebugger({ Key key, this.child }) : super(key: key);
+  final Widget child;
+  _SemanticsDebuggerState createState() => new _SemanticsDebuggerState();
+}
+
+class _SemanticsDebuggerState extends State<SemanticsDebugger> {
+  void initState() {
+    super.initState();
+    _SemanticsDebuggerClient.ensureInstantiated();
+    _SemanticsDebuggerClient.instance.addListener(_update);
+  }
+  void dispose() {
+    _SemanticsDebuggerClient.instance.removeListener(_update);
+    super.dispose();
+  }
+  void _update() {
+    setState(() {
+      // the generation of the _SemanticsDebuggerClient has changed
+    });
+  }
+  Point _lastTapLocation;
+  void _handlePointerDown(PointerDownEvent event) {
+    setState(() {
+      _lastTapLocation = event.position;
```

Comment:
Ah, yeah. Originally this was done with onTapDown but we don't always get that (e.g. for a drag if it's quick enough you don't get a tapdown). I should indeed rename the variable though.